caikit/runtime/servicers/training_management_servicer.py

```diff
@@ -32,16 +38,32 @@ class TrainingManagementServicerImpl:
     """This class contains the implementation of all of the RPCs that are required to run a
     service in Model Mesh as a Model-Runtime."""
 
-    def __init__(self):
-        self.training_manager = TrainingManager.get_instance()
+    # TODO: Align training status enums!
+    STATUS_MAPPING = {
+        ModelTrainerBase.TrainingStatus.QUEUED: TrainingStatus.NOT_STARTED,
+        ModelTrainerBase.TrainingStatus.RUNNING: TrainingStatus.PROCESSING,
+        ModelTrainerBase.TrainingStatus.COMPLETED: TrainingStatus.COMPLETED,
+        ModelTrainerBase.TrainingStatus.CANCELED: TrainingStatus.HALTED,
+        ModelTrainerBase.TrainingStatus.ERRORED: TrainingStatus.FAILED,
+    }
 
     def GetTrainingStatus(self, request, context):  # pylint: disable=unused-argument
         """Missing associated documentation comment in .proto file."""
         training_info = TrainingInfoRequest.from_proto(request)
 
-        return TrainingInfoResponse(
-            training_id=training_info.training_id,
-            status=self.training_manager.get_training_status(
-                training_info.training_id
-            ).value,
-        ).to_proto()
+        try:
+            model_future = MODEL_MANAGER.get_model_future(
+                training_id=training_info.training_id
+            )
+
+            return TrainingInfoResponse(
+                training_id=training_info.training_id,
+                status=self.STATUS_MAPPING[model_future.get_status()].value,
+            ).to_proto()
+        except ValueError as err:
+            raise CaikitRuntimeException(
+                grpc.StatusCode.NOT_FOUND,
+                "{} not found in the list of currently running training jobs".format(
+                    training_info.training_id,
+                ),
+            ) from err
```

Comment:
Is it time to implement a `def CancelTraining` here as well and put it in the training management api?

Reply:
Ooh, good call. Let's do that as a next story

Reply:
https://github.com/caikit/caikit/issues/332